bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -0,0 +1,276 @@
+        config = getConfigAs(SonnenConfiguration.class);
+        boolean validConfig = true;
+        String errors = "";
+        String statusDescr = null;
+        if (config.refreshInterval < 0 && config.refreshInterval > 999) {
+            errors += " Parameter 'refresh Rate' greater then 0 and less then 1000.";
+            statusDescr = "Parameter 'refresh Rate' greater then 0 and less then 1000.";
+            validConfig = false;
+        }
+        if (config.hostIP == null) {
+            errors += " Parameter 'hostIP' must be configured.";
+            statusDescr = "IP Address must be configured!";
+            validConfig = false;
+        }
+        errors = errors.trim();
+        Helper message = new Helper();
+        message.setStatusDescription(statusDescr);
+        if (validConfig) {
+            serviceCommunication.setConfig(config);
+            if (serviceCommunication.refreshBatteryConnection(message, this.getThing().getUID().toString())) {
+                updateStatus(ThingStatus.ONLINE);
+                updateLinkedChannels();
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+        }
+    }
+
+    /**
+     * Calls the service to update the battery data
+     *
+     * @param postdata
+     */
+    private boolean updatebatteryData(@Nullable String postdata) {
+        Helper message = new Helper();
+        if (serviceCommunication.refreshBatteryConnection(message, this.getThing().getUID().toString())) {
+            updateStatus(ThingStatus.ONLINE);
+            return true;
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    message.getStatusDesription());
+            return false;
+        }
+    }
+
+    private void updateLinkedChannels() {
+        verifyLinkedChannel(CHANNELBATTERYCHARGING);
+        verifyLinkedChannel(CHANNELBATTERYDISCHARGING);
+        verifyLinkedChannel(CHANNELGRIDFEEDIN);
+        verifyLinkedChannel(CHANNELCONSUMPTION);
+        verifyLinkedChannel(CHANNELSOLARPRODUCTION);
+        verifyLinkedChannel(CHANNELBATTERYLEVEL);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONBATTERY);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONGRID);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONPRODUCTION);
+        verifyLinkedChannel(CHANNELFLOWGRIDBATTERY);
+        verifyLinkedChannel(CHANNELFLOWPRODUCTIONBATTERY);
+        verifyLinkedChannel(CHANNELFLOWPRODUCTIONGRID);
+
+        if (!linkedChannels.isEmpty()) {
+            updatebatteryData(null);
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId());
+            }
```

Comment:
You do not need to run these lines as you are calling just after `startAutomaticRefresh` which will do exactly the same thing.

Reply:
ok

Reply:
You left the call to `updatebatteryData();`
To be removed.